src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -6711,6 +6762,67 @@ components:
           items:
             type: string
 
+    SnapshotAccessRequest:
+      type: object
+      properties:
+        name:
+          type: string
+        researchPurposeStatement:
+          type: string
+        datasetRequest:
+          $ref: '#/components/schemas/SnapshotRequest'
+      description: Request for access to a snapshot.
+
+    SnapshotRequest:
+      type: object
+      properties:
+        cohorts:
+          type: array
+          items:
+            $ref: '#/components/schemas/Cohort'
```

Comment:
Nit: I would name this `SnapshotBuilderCohort`. I think that will make it easily searchable and clear that it is relating to the snapshot builder when looking at the yaml.